makefiles/tools/compile_commands.inc.mk

```diff
@@ -0,0 +1,82 @@
+# clangd doesn't automatically include system headers of embedded toolchain. Add this as extra
+# include dir, but allow overriding it
+COMPILE_COMMANDS_INCLUDES_EXTRA ?= -I/usr/$(TARGET_ARCH)/include
+# clangd doesn't support -Wformat-truncation. Just ignore unknown warnings here
+COMPILE_COMMANDS_CFLAGS_FLAGS   ?= -Wno-unknown-warning-option
+# The following list of flags confuse clangd
+COMPILE_COMMANDS_STRIP_FLAGS    ?= -mno-thumb-interwork
+
+# Templates used to generate valid JSON. The contain magic keywords in pointy braces to be replaced
+# by the correct values
+define T_COMPILE_COMMAND
+  {
+    "arguments": [
+      "<COMPILER>",
+ <FLAGS>      "-o",
+      "<OUTPUT>",
+      "-c",
+      "<FILE>"
+    ],
+    "directory": "<DIRECTORY>",
+    "file": "<DIRECTORY>/<FILE>",
+    "output": "<OUTPUT>"
+  }<SEPARATOR>
+
+endef
+
+define T_COMPILE_COMMANDS
+[
+<COMMANDS>]
+endef
+
+define T_FLAGS
+     "<FLAG>",
```

Comment:
It is entirely valid syntax, but the JSON files become quite long when each flag is on a separate line when the module has more than 1-2 source files.

Reply:
I'd prefer to stick with it. This intentionally generates the exact same layout `bear` does. And we have like 30 flags or so, having them all in a single line is trouble to navigate.